### ISSUE TITLE
Support ls

### ISSUE THREAD

**Issue by eyalb181** [metalbear-co/mirrord#120]
Our files feature currently support most of the relevant syscalls, but we still need to add support for ls.

After #904, this is what is left to be done (prioritized)

- [x]  Go Linux x86-64 (same flow/integration test should work~)
- [x]  Python list directory
- [x]  Node list directory
- [ ] `ls` binary - macOS
- [ ] `ls` binary - Linux
- [ ]  All of the libc functions:
  - [X] opendir
  - [X] fdopendir
  - [X] readdir
  - [X] readdir_r
  - [ ] telldir
  - [ ] seekdir
  - [ ] rewinddir
  - [X] closedir
  - [x] dirfd

I think the best way would be to drop each as separate PRs, first few tasks will resolve most of the libc functions :)

**Comment by infiniteregrets** [comment]
openat was fixed in #151, detours for ls still pending but low priority

**Comment by t4lz** [comment]
Maybe we should recheck some of the use-cases after #2003. Maybe more stuff works now?

**Comment by Razz4780** [comment]
Ah yes, I checked `ls` on linux after #2003
It worked :)

**Comment by t4lz** [comment]
@MartinEmrich files under `/etc/` are read locally by default. In order to make mirrord read that file remotely even though it is in `/etc/`, you can add the file's path to the [`feature.fs.read_only`](https://mirrord.dev/docs/overview/configuration/#feature-fs-read_only) configuration value, or to `feature.fs.read_write`, if you also want to write to that file. Feel free to come to our [discord](https://discord.gg/metalbear) for live support.

**Comment by MartinEmrich** [comment]
@t4lz Thanks, that did the trick, it works!

**Comment by eyalb181** [comment]
@infiniteregrets Could you please retest the first four use cases (Python list - ls on Linux)?

**Comment by infiniteregrets** [comment]
macos:
- does not work on 
python: i used os.walkdir and scandir
node: fs.readdir

same for ls, however I am assuming to make ls work we'd have to implement the fts API: https://opensource.apple.com/source/Libc/Libc-825.40.1/gen/fts.c.auto.html

**Comment by infiniteregrets** [comment]
support for ls in python and node was added by https://github.com/metalbear-co/mirrord/pull/2145
